en_US/changes/changes-ee-v5.md

```diff
@@ -1,5 +1,250 @@
+  - `emqx_subscriptions_shared_max`
+
+  Additionally, this fix rectified an issue within the `/stats` endpoint concerning the `subscriptions.shared.count` and `subscriptions.shared.max` fields. Previously, these values failed to update promptly following a client's disconnection or unsubscription from a Shared-Subscription.
+
+- [#12390](https://github.com/emqx/emqx/pull/12390) Fixed an issue where the `/license` API request may crash during cluster joining processes.
+
+- [#12411](https://github.com/emqx/emqx/pull/12411) Fixed a bug where `null` values would be inserted as `1853189228` in `int` columns in Cassandra data integration.
+
+- [#12522](https://github.com/emqx/emqx/pull/12522) Refined the parsing process for Kafka bootstrap hosts to exclude spaces following commas, addressing connection timeouts and DNS resolution failures due to malformed host entries.
+
+- [#12656](https://github.com/emqx/emqx/pull/12656) Implemented a topic verification step for creating GCP PubSub Producer actions, ensuring failure notifications when the topic doesn't exist or provided credentials lack sufficient permissions.
+
+- [#12678](https://github.com/emqx/emqx/pull/12678) Enhanced the DynamoDB connector to clearly report the reason for connection failures, improving upon the previous lack of error insights.
+
+- [#12681](https://github.com/emqx/emqx/pull/12681) Fixed a security issue where secrets could be logged at debug level when sending messages to a RocketMQ bridge/action.
+
+- [#12715](https://github.com/emqx/emqx/pull/12715) Fixed a crash that could occur during configuration updates if the connector for the ingress data integration source had active channels.
+
+- [#12740](https://github.com/emqx/emqx/pull/12740) Fixed an issue when durable sessions could not be kicked out.
+
+- [#12767](https://github.com/emqx/emqx/pull/12767) Fixed issues encountered during upgrades from 5.0.1 to 5.5.1, specifically related to Kafka Producer configurations that led to upgrade failures. The correction ensures that Kafka Producer configurations are accurately transformed into the new action and connector configuration format required by EMQX version 5.5.1 and beyond.
+
+- [#12768](https://github.com/emqx/emqx/pull/12768) Addressed a startup failure issue in EMQX version 5.4.0 and later, particularly noted during rolling upgrades from versions before 5.4.0. The issue was related to the initialization of the routing schema when both v1 and v2 routing tables were empty.
+
+  The node now attempts to retrieve the routing schema version in use across the cluster instead of using the v2 routing table by default when local routing tables are found empty at startup. This approach mitigates potential conflicts and reduces the chances of diverging routing storage schemas among cluster nodes, especially in a mixed-version cluster scenario. 
+
+  The version also logs instructions for how to manually resolve if conflict is detected in a running cluster.
```

Comment:
@id , where can users view the logged instructions? Are there in the log file?

Reply:
@Meggielqk [instructions](https://github.com/emqx/emqx/pull/12768/files#diff-c5fccef62b11c07952ba90fc2743e5a587d575eee16858c1bac11113343168b8R804) are logged together with the error message via [configured log handlers with severity "Critical"](https://github.com/emqx/emqx/pull/12768/files#diff-c5fccef62b11c07952ba90fc2743e5a587d575eee16858c1bac11113343168b8R770) and on [standard error](https://github.com/emqx/emqx/pull/12768/files#diff-c5fccef62b11c07952ba90fc2743e5a587d575eee16858c1bac11113343168b8R769) to make sure this message will not get lost even if no log handlers are configured.

Reply:
Instructions are not available separately because they are generated in runtime using information about the current cluster state.

Reply:
Thank you for the answer. I just wonder if users will know how to access the instructions. Do you think we should mention at the end: "You can view the instructions in the log outputs."?

Reply:
This is implied by the "logs instructions" phrase.
May be rephrase it like this:

```suggestion
  If conflict is detected in a running cluster, EMQX writes instructions on how to manually resolve it in the log as part of the error message with `critical` severity. The same error message and instructions will also be written on standard error to make sure this message will not get lost even if no log handler is configured.
```